Pixel Game/Assets/Scripts/NpcData.cs

```diff
@@ -15,16 +15,16 @@ public class NpcData : MonoBehaviour
 
 
 
-    private GameObject dialogueController;
-    private GameObject dataToPassGameObject;
+    private DialogueController dialogueController;
+    private DataToPassBetweenScenes dataToPass;

```

Comment:
Is there a specific reason to get the class rather than the gameobject in the scene? @yousernaym

Reply:
The GameObjects were never used anyway. It was always `dataToPassGameObject.GetComponent<DataToPassBetweenScenes>()`for example. I just moved those calls to Start()